packages/ui/cypress/component/SDatePicker.spec.cy.ts

```diff
@@ -0,0 +1,194 @@
+import { mount } from '@cypress/vue'
+import { config } from '@vue/test-utils'
+import { SDatePicker } from '@/lib'
+
+before(() => {
+  config.global.components = { SDatePicker }
+})
+
+after(() => {
+  config.global.components = {}
+})
+
+describe('SDatePicker', () => {
+  const getEl = (id: string) => cy.get(`[data-cy=${id}]`)
+  const picker = () => cy.get(`[data-testid=date-picker]`)
+
+  it('Time prop works correctly', () => {
+    mount({
+      template: `
+        <SDatePicker v-model="date" :type="'day'" :time="true" data-cy="picker"/>
+        <p data-cy="result">{{date}}</p>
+      `,
+      setup() {
+        const date = ref(new Date())
+        return {
+          date,
+        }
+      },
+    })
+
+    picker().children('.s-date-picker-time-panel').should('exist')
+  })
+
+  it('Displayed two calendars in range mode', () => {
+    mount({
+      template: `
+        <SDatePicker v-model="date" :type="'range'"/>
+        <p data-cy="result">{{date}}</p>
+      `,
+      setup() {
+        const date = ref(new Date())
+        return {
+          date,
+        }
+      },
+    })
+    picker().children('.s-date-picker-calendars-panel').children().should(`have.length`, 2)
+  })
+
+  it('Displayed one calendar in day mode', () => {
+    mount({
+      template: `
+        <SDatePicker v-model="date" :type="'day'"/>
+        <p data-cy="result">{{date}}</p>
+      `,
+      setup() {
+        const date = ref(new Date())
+        return {
+          date,
+        }
+      },
+    })
+    picker().children('.s-date-picker-calendars-panel').children().should(`have.length`, 1)
+  })
+
+  it('No options panel in Pick mode', () => {
+    mount({
+      template: `
+        <SDatePicker v-model="date" :type="'pick'"/>
+        <p data-cy="result">{{date}}</p>
+      `,
+      setup() {
+        const date = ref([])
+        return {
+          date,
+        }
+      },
+    })
+    picker().children('.s-date-picker-options-panel').should('not.exist')
+  })
+
+  describe('Panels tests', () => {
+    const clickDoneButton = () => cy.get(`[data-testid=date-picker-done-button]`).click()
+    it('Input panel works correctly', () => {
+      mount({
+        template: `
+          <SDatePicker v-model="date" :type="'day'" data-cy="picker"/>          
+          <p data-cy="result">
+            <span data-cy="date">{{date.getDate()}}</span>
+            <span data-cy="month">{{date.getMonth()}}</span>
+            <span data-cy="year">{{date.getFullYear()}}</span>
+          </p>
+        `,
+        setup() {
+          const date = ref(new Date())
+          return {
+            date,
+          }
+        },
+      })
+      getEl('picker').children().first().click()
+      picker().get('.s-date-picker-options-panel__item').last().click()
+      picker().get('.s-date-picker-custom-panel__input').focus().clear().type('01012000').blur()
+      cy.get(`[data-testid=date-picker-done-button]`).click({ force: true })
```

Comment:
It is better to move hardcoded classes out of tests to constants, like:

```ts
const OPTIONS_PANEL_ITEM = '.s-date-picker-options-panel__item'
```

As for `[data-testid=xxx]`, it should be abstracted with a factory as well (I beleive there is some in a shared test utils module).